src/client/Contracts.js

```diff
@@ -928,4 +928,250 @@ exports.ResetTenantId = async function({contractAddress, objectId, versionHash})
   }
 };
 
+/**
+ * Enum for object types that can be cleaned up after object deletion.
+ * Used by the ObjectCleanup method to determine which associated objects to clean.
+ *
+ * @property {string=} LIBRARY - Cleanup libraries
+ * @property {string=} CONTENT_OBJECT - Cleanup content objects
+ * @property {string=} GROUP - Cleanup access groups
+ * @property {string=} CONTENT_TYPE - Cleanup content types
+ * @property {string=} ALL - Cleanup all of the above
+ */
+const ObjectTypesToClean = Object.freeze({
+  LIBRARY: "library",
+  CONTENT_OBJECT: "content_object",
+  GROUP: "group",
+  CONTENT_TYPE: "content_type",
+  ALL: "all"
+});
+
+/**
+ * Cleans up objects (libraries, content objects, groups or content types)
+ * associated with a given user or object
+ *
```

Comment:
Cleans up deleted objects pointed to by the access index of a given "access group" or "user wallet"

Contracts of type "access group" and "user wallet" contain an "access index"  - a list of objects that they have access to.

There are 4 specific indexes, one for each object type:
    - content
    - library
    - access groups
    - content types

If an object gets deleted and the access index still points to it, it will cause errors in API calls for the access group or user wallet.

This function checks each index for objects that are deleted, and removes them from the index (either all indexes or just the one specified by parameter 'objectTypeToClean')